lib/items.ts

```diff
@@ -2,45 +2,48 @@ import { Item, PlayedItem } from "../types/item";
 import { createWikimediaImage } from "./image";
 
 export function getRandomItem(deck: Item[], played: Item[]): Item {
-  const playedYears = played.map((item): number => {
-    return item.year;
-  });
-
   const periods: [number, number][] = [
-    [-100000, 1400],
-    [1400, 1850],
-    [1850, 1930],
-    [1930, 2020],
+    [-100000, 1000],
+    [1000, 1800],
+    [1800, 2020],
   ];
   const [fromYear, toYear] =
     periods[Math.floor(Math.random() * periods.length)];
   const avoidPeople = Math.random() > 0.5;
+  let distance = 5;
+  if (played.length < 11) { 
+    distance = 110 - 10 * played.length;
+  } else if (played.length > 40) {
+    distance = 1;
+  }
+  let item = deck[Math.floor(Math.random() * deck.length)];
 
-  const candidates = deck.filter((candidate) => {
-    if (avoidPeople && candidate.instance_of.includes("human")) {
-      return false;
+  for (let i = 0; i < 10000; i++) {
```

Comment:
This is undoing the changes made in https://github.com/tom-james-watson/wikitrivia/pull/22. I prefer the cleanness of pre-filtering an array, personally. We decided the performance impact is negligible. I think it's best to revert back to the previous way.

Reply:
Roger, the discussion on #22 makes sense.  This has been reverted.